### PR TITLE
feat: document SD_SOURCE_PATH environment variable

### PR DESCRIPTION
**Pull request by jithine**

### CODE REVIEW

**Review by tkyi — APPROVED** [review]

File: docs/user-guide/configuration/sourcePaths.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ jobs:
 
 In this example, the job `main` will be triggered if there are any changes to files under `src/app/` or the `screwdriver.yaml` file (like on `src/app/main.js`, `src/app/package.json`, etc.). The `main` job will **not**, however, be triggered on changes to `README.md`, `test/`, or `src/other/`.
 
+### Matched source path
+
+Screwdriver will expose the matched souce path in an environment variable `SD_SOURCE_PATH`. This can be used to write scripts which depends on the source path that triggered the build.
```
Comment (tkyi) [inline docs/user-guide/configuration/sourcePaths.md]: source* path

Comment (tkyi) [reply docs/user-guide/configuration/sourcePaths.md]: ~the matched souce path~`the source path(s) that triggered this build to run`?

not sure what makes more sense

File: docs/user-guide/environment-variables.md [hunk 2]
```diff
@@ -44,6 +44,7 @@ _Note: Environment variables set in one job cannot be accessed in another job. T
 |------|-------|
 | SD_SOURCE_DIR | Location of checked-out code |
 | SD_ARTIFACTS_DIR | Location of built/generated files |
+| SD_SOURCE_PATH | Location of matched source path. See [Source Paths](./configuration/sourcePaths). |
```
Comment (tkyi) [inline docs/user-guide/environment-variables.md]: ~Location of matched source path~`The source path(s) that triggered this build to run`?

File: docs/user-guide/configuration/sourcePaths.md (hunk 1 above)
Comment (minzcmu) [inline docs/user-guide/configuration/sourcePaths.md]: Based on the logic in model, SD_SOURCE_PATH will be the FIRST match path. It is possible that a PR can multiple files match different source defined. Should we say anything about that in the doc?

File: docs/user-guide/environment-variables.md (hunk 2 above)
Comment (jithine) [reply docs/user-guide/environment-variables.md]: @tkyi it will be only one value, so `path(s)` is not correct. I will update the rest.

**Review by minzcmu — APPROVED** [review]

File: docs/user-guide/environment-variables.md (hunk 2 above)
Comment (tkyi) [reply docs/user-guide/environment-variables.md]: Why will it only be one value? Users can specify multiple source paths in their yaml, and it's possible for multiple source paths to match, right?

File: docs/user-guide/configuration/sourcePaths.md (hunk 1 above)
Comment (tkyi) [reply docs/user-guide/configuration/sourcePaths.md]: Should it return *all* matches vs just the *first*?

**Pull request merged**